dashboard/src/main/java/com/google/cloud/tools/opensource/dashboard/DashboardMain.java

```diff
@@ -375,23 +390,68 @@ private static ArtifactResults generateArtifactReport(
     return upperBoundFailures;
   }
 
+  /**
+   * Returns a table where rows are jar files in {@code jarFilter}, columns are {@link
```

Comment:
I wonder if this table is as helpful as it should be. You're partitioning the problems multimap by jar file, right? And as far as I can tell in the templates you're iterating over the jars. With the table, you're getting a `Map<SymbolProblem, ImmutableSet<String>>` for each jar row key.

If this method returned an `ImmutableMap<Path, ImmutableSetMultimap<SymbolProblem, ClassFile>>`, then you'd get an `ImmutableSetMultimap<SymbolProblem, ClassFile>` for each jar key, which you could then invert as you needed to, or `forEach` over, or what have you.

And the implementation of this method would become very simple. Something like:

```java
return ImmutableMap.copyOf(
    Maps.transformValues(
        Multimaps.index(symbolProblems.entrySet(), entry -> entry.getValue().getJar()).asMap(), 
        ImmutableSetMultimap::copyOf));
```

Something to consider.

Reply:
Trying that approach.

Reply:
Now `indexByJar` returns map of multimap.